src/core/lib/support/alloc_new.h

```diff
@@ -0,0 +1,30 @@
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPC_SUPPORT_ALLOC_NEW_H
+#define GRPC_SUPPORT_ALLOC_NEW_H
+
+#include <grpc/support/alloc.h>
+
+#define GPR_NEW(expr) new (gpr_zalloc) expr
```

Comment:
I would prefer either having two macros one with gpr_zalloc and one with gpr_malloc, or simply using gpr_malloc and setting the fields in a constructor.

Reply:
not applicable anymore